pkg/microservice/aslan/core/environment/service/environment.go

```diff
@@ -726,9 +767,40 @@ func CreateHelmProduct(productName, userName, requestID string, args []*CreateHe
 			if serviceTmpl.Type == setting.K8SDeployType || serviceTmpl.Type == setting.HelmDeployType {
 				serviceResp.Containers = make([]*commonmodels.Container, 0)
 				for _, c := range serviceTmpl.Containers {
+					image := c.Image
+					for _, rc := range productObj.ChartInfos {
+						if rc.ServiceName != serviceTmpl.ServiceName {
+							continue
+						}
+						mergeYaml, err := helmtool.MergeOverrideValues(rc.ValuesYaml, defaultValuesYaml, rc.GetOverrideYaml(), rc.OverrideValues)
+						if err != nil {
+							errMsg := fmt.Sprintf("Failed to MergeOverrideValues for product template %s,error:%s", productObj.ProductName, err)
+							log.Error(errMsg)
+							return e.ErrCreateEnv.AddDesc(errMsg)
+						}
+						mergedValuesYamlFlattenMap, err := converter.YamlToFlatMap([]byte(mergeYaml))
+						if err != nil {
+							errMsg := fmt.Sprintf("mergeYamlToFlatMap product template %s,error:%s", productObj.ProductName, err)
+							log.Error(errMsg)
+							return e.ErrCreateEnv.AddDesc(errMsg)
+						}
+						imageRepo, ok1 := mergedValuesYamlFlattenMap[c.ImagePath.Repo]
+						imageTag, ok2 := mergedValuesYamlFlattenMap[c.ImagePath.Tag]
```

Comment:
use commonservice.GeneImageURI to package full image url

Reply:
GeneImageURI also needs to unwrap the image to make sure it's complete. So it's not easier